rpcs3/Emu/Cell/Modules/cellPad.cpp

```diff
@@ -235,17 +235,25 @@ s32 cellPadGetData(u32 port_no, vm::ptr<CellPadData> data)
 		btnChanged = true;
 	}
 
-	//not sure if this should officially change with capabilities/portsettings :(
-	data->len = 24;
-
-	if (pad.m_buffer_cleared)
+	// the real hardware only fills the buffer up to "len" elements (16 bit each)
+	if (pad.m_port_setting & CELL_PAD_SETTING_SENSOR_ON)
+	{
+		// there is always valid data in the buffer for this mode
+		data->len = 24;
```

Comment:
This is on the right track but not sure it should be here in cellPad. The issue is due to the fact the 'padHandler's (xinput/keyboard/mmjoy) don't actually emulate gyro packets correctly. This issue is largely gone with ds4 because it actually reports gyro/accel packets correctly

Reply:
This should be a separate issue. If a game enables the gyro in the current master it also gets the length = 24 packet with fake (centered) gyro data. Maybe the gyro could be emulated with mouse input if the primary input device does not have a gyro?